### PR TITLE
Add Operation to access _all_dbs

### LINKED ISSUE #28: Support the `_all_dbs` endpoint
Support the `_all_dbs` endpoint, see [CouchDB docs](http://couchdb.readthedocs.org/en/1.6.1/api/server/common.html#all-dbs) for more information.

### PR DESCRIPTION
**Pull request by rhyshort**
## What
Added API to list all databases on the remote server.
## How
Added new `GetAllDatabasesOperation` .

## Reivewers

reviewer @emlaver

## Issues

Resolves #28

### CODE REVIEW

File: Source/GetAllDatabasesOperation.swift [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ }
+ allDbs.completionHandler = { (response, httpInfo, error) in 
+    if let error = error {
+        // handle the error case
+    }
+    // process the response information
+ }
+ 
+ - Note: The `response` parameter for the `completionHandler` **will not** match the response
+  reccived from Couch. This is because CouchDB returns an array as a top level object and the
```
Comment (emlaver) [inline Source/GetAllDatabasesOperation.swift]: change `reccived` to `received`.

Comment (rhyshort) [inline Source/GetAllDatabasesOperation.swift]: fixed in 2b4f621

File: Source/GetAllDatabasesOperation.swift [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ 
+ - Note: The `response` parameter for the `completionHandler` **will not** match the response
+  received from Couch. This is because CouchDB returns an array as a top level object and the
+  completion handler only accepts Dictionaries. To access the list of database names, use the key
+  `databases`
+ 
+ */
+public class GetAllDatabasesOperation : CouchOperation {
+    
+    public var databaseHandler: ((databaseName: String) -> Void)?
```
Comment (emlaver) [inline Source/GetAllDatabasesOperation.swift]: Could you add documentation similar to what's in `FindDocumentsOperation`:
https://github.com/cloudant/swift-cloudant/blob/master/Source/FindDocumentsOperation.swift#L141

Comment (rhyshort) [inline Source/GetAllDatabasesOperation.swift]: done in f9831fb

**Comment by emlaver** [comment]
LGTM +1

**Pull request merged**